src/java.base/share/classes/java/lang/reflect/code/bytecode/BytecodeGenerator.java

```diff
@@ -1110,23 +1111,43 @@ private boolean needToAssignBlockArguments(Block.Reference ref) {
     }
 
     private void assignBlockArguments(Block.Reference ref) {
+        Block target = ref.targetBlock();
         List<Value> sargs = ref.arguments();
-        List<Block.Parameter> bargs = ref.targetBlock().parameters();
-        // First push successor arguments on the stack, then pop and assign
-        // so as not to overwrite slots that are reused slots at different argument positions
-        boolean jumpingToCatchBlock = catchingBlocks.get(ref.targetBlock().index());
-        for (int i = 0; i < bargs.size(); i++) {
-            Block.Parameter barg = bargs.get(i);
-            Value value = sargs.get(i);
-            if (!barg.uses().isEmpty() && !barg.equals(value)) {
+        if (catchingBlocks.get(target.index())) {
+            // Jumping to an exceptio handler, exception parameter is expected on stack
+            Value value = sargs.getFirst();
+            if (oprOnStack == value) {
+                oprOnStack = null;
+            } else {
+                load(value);
+            }
+        } else if (target.predecessors().size() > (target.isEntryBlock() ? 0 : 1)) {
```

Comment:
An entry block cannot have any predecessors. Are you observing that? If so its a bug.

Reply:
Right, it does not happen.
I added it intuitively as looping to bci 0 is frequent in the bytecode.

Reply:
Interesting. I suppose when lifting such cases you have to create a block (with possible parameters) that the entry block unconditionally branches too.

Reply:
Yes, in such cases an explicit method entry stack map frame triggers a new block with implicit branch from the entry block.